src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -508,6 +508,7 @@ TransactionManager::rewrite_extent_ret TransactionManager::rewrite_extent(
       return rewrite_extent_iertr::now();
     }
     extent = updated;
+    ceph_assert(!extent->is_pending_io());
```

Comment:
Maybe we should drop `cache->update_extent_from_transaction(t, extent)` instead if this `ceph_assert()` doesn't break anything.

Reply:
It looks to me that `cache->update_extent_from_transaction()` can detect whether an extent is already retired and shouldn't be rewritten, so I think we probably need to keep it.